CentralityiOS/CentralityiOS/ViewControllers/UsageDashboardViewController.m

```diff
@@ -0,0 +1,177 @@
+    
+    [self setupPieChartView:self.pieChart];
+    self.pieChart.delegate = self;
+    ChartLegend *l = self.pieChart.legend;
+    l.horizontalAlignment = ChartLegendHorizontalAlignmentRight;
+    l.verticalAlignment = ChartLegendVerticalAlignmentTop;
+    l.orientation = ChartLegendOrientationVertical;
+    l.drawInside = NO;
+    l.xEntrySpace = 7.0;
+    l.yEntrySpace = 0.0;
+    l.yOffset = 0.0;
+    
+    // entry label styling
+    self.pieChart.entryLabelColor = UIColor.blackColor;
+    self.pieChart.entryLabelFont = [UIFont fontWithName:@"HelveticaNeue-Light" size:12.f];
+    
+    
+    
+    [self.pieChart animateWithXAxisDuration:1.4 easingOption:ChartEasingOptionEaseOutBack];
+}
+
+- (void)loadChartWithCategories{
+    [[CentralityHelpers queryForUsersCategories] findObjectsInBackgroundWithBlock:^(NSArray * _Nullable categoriesFromQuery, NSError * _Nullable error) {
+        self.userCategories = [categoriesFromQuery mutableCopy];
+        [self updateChartData];
+    }];
+}
+
+- (void)setupPieChartView:(PieChartView *)chartView
+{
+    chartView.usePercentValuesEnabled = YES;
+    chartView.drawSlicesUnderHoleEnabled = NO;
+    chartView.holeRadiusPercent = 0.58;
+    chartView.transparentCircleRadiusPercent = 0.61;
+    chartView.chartDescription.enabled = NO;
+    [chartView setExtraOffsetsWithLeft:5.f top:10.f right:5.f bottom:5.f];
+    
+    chartView.drawCenterTextEnabled = YES;
+    
+    NSMutableParagraphStyle *paragraphStyle = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
+    paragraphStyle.lineBreakMode = NSLineBreakByTruncatingTail;
+    paragraphStyle.alignment = NSTextAlignmentCenter;
+    
+    NSMutableAttributedString *centerText = [[NSMutableAttributedString alloc] initWithString:@"Tasks\nby Category"];
+    [centerText setAttributes:@{
+        NSFontAttributeName: [UIFont fontWithName:@"HelveticaNeue-Light" size:11.f],
+        NSParagraphStyleAttributeName: paragraphStyle
+    } range:NSMakeRange(0, centerText.length)];
+    chartView.centerAttributedText = centerText;
+    
+    chartView.drawHoleEnabled = YES;
+    chartView.rotationAngle = 0.0;
+    chartView.rotationEnabled = YES;
+    chartView.highlightPerTapEnabled = YES;
+    
+    ChartLegend *l = chartView.legend;
+    l.horizontalAlignment = ChartLegendHorizontalAlignmentRight;
+    l.verticalAlignment = ChartLegendVerticalAlignmentTop;
+    l.orientation = ChartLegendOrientationVertical;
+    l.drawInside = NO;
+    l.xEntrySpace = 7.0;
+    l.yEntrySpace = 0.0;
+    l.yOffset = 0.0;
+}
+
+- (void)updateChartData
+{
+    if (self.shouldHideData)
+    {
+        self.pieChart.data = nil;
+        return;
+    }
+    
+    [self setDataCount:self.userCategories.count];
+}
+
+- (void)setDataCount:(NSUInteger)count
+{
+    NSMutableArray *values = [[NSMutableArray alloc] init];
+    for (int i = 0; i < count; i++)
+    {
+        CategoryObject* currentCategory = self.userCategories[i % self.userCategories.count];
```

Comment:
i'm confused by this! why are you doing `i % count`? is this just to prevent indexing outside of the array, or am i missing something?

if you're just trying to stay in the correct range, i'd suggest instead setting count to `min(count, self.userCategories.count)` - unless you want it to circle around the categories and potentially display some of them more than once, which i think is what will happen now?

Reply:
I can't remember exactly why I initially did this. I think it may have had something to do with modifying the original demo example chart that I received from the charts pod. For now I'm just going to write self.userCategories[i]